src/EditorFeatures/Core/KeywordHighlighting/HighlighterViewTaggerProvider.cs

```diff
@@ -100,7 +100,7 @@ protected override ITaggerEventSource CreateEventSource(ITextView textView, ITex
             var onExistingTags = context.HasExistingContainingTags(new SnapshotPoint(snapshot, position));
             if (onExistingTags)
             {
-                context.SetSpansTagged(ImmutableArray<SnapshotSpan>.Empty);
+                context.ClearSpansActuallyTagged();
```

Comment:
i despise this.  We really need to just change the tagging API to allow the engine to just ask "should i just keep the tags from before?".  Highlighting can answer that directly, and the engine will do the right thing (versus trying to infer from this piece of data passed along like this).

Reply:
Validating that this experience is still good.